presto-main/src/test/java/com/facebook/presto/operator/project/TestDictionaryAwarePageProjection.java

```diff
@@ -154,43 +188,77 @@ private static DictionaryBlock createDictionaryBlockWithUnusedEntries(int dictio
         return new DictionaryBlock(dictionary, ids);
     }
 
-    private static void testProject(Block block, Class<? extends Block> expectedResultType)
+    private static Block projectWithYield(PageProjectionOutput output, DriverYieldSignal yieldSignal)
+    {
+        for (int i = 0; i < 10; i++) {
+            yieldSignal.setWithDelay(1, executor);
+            yieldSignal.forceYieldForTesting();
```

Comment:
This doesn't look like useful test. Yielding at the beginning essentially means that each `compute` call will do nothing and return almost immediately. I assume that the reason you wrote this test is to make sure that your code will produce correct results regardless of where it got forced a yield.

I feel like a YieldSignal implementation that tosses a coin after every isSet call should be helpful here.